tests/api/testcl.hpp

```diff
@@ -510,12 +517,19 @@ class WithCommandQueue : public WithContext {
 #ifndef COMPILER_AVAILABLE
         GTEST_SKIP();
 #endif
-        WithContext::SetUp();
         auto queue = CreateCommandQueue(device(), properties);
         m_queue = queue.release();
     }
 
-    void SetUp() override { SetUpQueue(0); }
+    void SetUp() override {
+        WithContext::SetUp();
+        SetUpQueue(0);
+    }
+
+    void SetUpWithQueueProperties(const cl_context_properties* _prop) {
```

Comment:
`SetUpWithQueueProperties` indicates that `_prop` is a queue properties, but it is used for the context properties, it does not make sense.
The idea here is to have a `WithCommandQueue::WithSetUpWithQueueProperties` that can be used in `WithProfiledCommandQueue::SetUp` instead of have to call `WithCommandQueue::SetUpQueue` explicitely.

Reply:
We will also need a `WithCommandQueue::SetUpWithContextProperties`